### PR TITLE
Also ensure filewatcher subscriptions for unresolved analyzers

### PR DESCRIPTION
**Pull request by Pilchie**
**Customer scenario**: When opening a project where the analyzer doesn't exist on disk, we may not have subscribed to file change tracking when it starts existing, so we may not add it.
**Bugs this fixes:** Fixes #20149 / https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20IDE/_workitems?id=460373&_a=edit
**Workarounds, if any**: Wait long enough after opening a project before restoring.
**Risk**: Low - just restoring the wait that was previously there.
**Performance impact**: Small - may lose some of the benefit of https://github.com/dotnet/roslyn/pull/20149, in the case where the files don't exist, but still better than 15.2
**Is this a regression from a previous update?**: Yes, https://github.com/dotnet/roslyn/pull/20149 introduced this.
**Root cause analysis:** We missed it in code review of the previous fix.
**How was the bug found?**: Code inspection.

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs [hunk 1]
```diff
@@ -135,5 +136,37 @@ public Assembly LoadFromPath(string fullPath)
...
+
+            public override string FullPath
+                => _underlying.FullPath;
+
+            public override object Id
+                => _underlying.Id;
+
+            public override ImmutableArray<DiagnosticAnalyzer> GetAnalyzers(string language)
+            {
+                _visualStudioAnalyzer._tracker.EnsureSubscription();
```
Comment (rchande) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs]: This looks so wrong (member name that starts with underscore after dot).

Comment (Pilchie) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs]: Nested types are tricky like that.  Note that this is the same pattern as `AnalyzerAssemblyLoaderThatEnsuresFileBeingWatched` lower in the file.

**Comment by Pilchie** [comment]
Ping @jasonmalinowski for review.

**Comment by Pilchie** [comment]
@dotnet/roslyn-infrastructure - I can't seem to find the test log or screenshots or anything in the artifacts folder for https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_debug_vs-integration_prtest/23/ ?

**Comment by Pilchie** [comment]
Nevermind - this was the uninstall failure.

**Comment by Pilchie** [comment]
retest windows_debug_vs-integration_prtest please

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs [hunk 2]
```diff
@@ -135,5 +136,37 @@ public Assembly LoadFromPath(string fullPath)
                 return _analyzer._loader.LoadFromPath(fullPath);
             }
         }
+
+        /// <summary>
+        /// This custom <see cref="AnalyzerReference"/>, just wraps an existing <see cref="UnresolvedAnalyzerReference"/>,
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs]: For some reason the word "existing" here threw me for a loop -- it's still the thing creating the underlying one too.

**Comment by Pilchie** [comment]
Tag @MattGertz (also marked VSO bug).

**Comment by MattGertz** [comment]
@Pilchie I already sent today's crop to JoC, so this will be yours to drive.  I'm supportive of the fix.

**Pull request merged**